### PR TITLE
Add support for Optec FocusLynx based focus controllers

### PR DESCRIPTION
**Pull request by AnthonyHorton**
Includes support for the  Starlight Instruments Focus Boss II controller and Feather Touch Focuser, as used by Huntsman.

This is an initial implementation incorporating all the standard Focuser features required for current POCS operations, but not the advanced features such as automated temperature compensation, support for 2 focus motors on one controller or control over Ethernet or WiFi. It also lacks, for now, the device node scan that most of the current Camera & Focuser classes in POCS have. Instead the (possibly non-deterministic) serial port device node needs to be manually specified in the config file.

I'm putting the PR in now because there will Huntsman observing this week and I want to test this then. I'll write up Issues for the above enhancements.

### CODE REVIEW

**Comment by AnthonyHorton** [comment]
Addresses https://github.com/panoptes/POCS/issues/432

**Comment by AnthonyHorton** [comment]
[RASA focuser prep](https://trello.com/c/k0CeuwYt/20-rasa-focuser-prep)

File: pocs/focuser/focuslynx.py [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+            return self._position
+        else:
+            return position
+
+    def halt(self):
+        """
+        Causes the focuser to immediately stop any movements
+        """
+        self._send_command('<F{:1d}HALT>'.format(self._focuser_number), command_type='HALT')
+        self.logger.warning("Focuser {} halted".format(self.uid))
```
Comment (wtgee) [inline pocs/focuser/focuslynx.py]: Maybe need to run `_update_focuser_status()` after this?

File: pocs/focuser/focuser.py [hunk 2]
```diff
@@ -138,7 +138,7 @@ def move_to(self, position):
 
     def move_by(self, increment):
```
Comment (wtgee) [inline pocs/focuser/focuser.py]: Is the `increment` variable just a count? Maybe some docstring here since it's in base class now.

Comment (AnthonyHorton) [reply pocs/focuser/focuser.py]: Yeah, it's just the number of steps to change the focuser position by. Could do with a docstring though. There are a few other places where it's not properly documented too.

File: pocs/focuser/focuslynx.py (hunk 1 above)
Comment (AnthonyHorton) [reply pocs/focuser/focuslynx.py]: Probably should, yes.  Kind of inclined to `warning.warn()` here too, as I basically see `halt()` being the emergency stop button, which should have plenty of user feedback.

**Comment by AnthonyHorton** [comment]
@lspitler Did you get a chance to test this at the observatory on your recent trip?

**Comment by wtgee** [comment]
> @lspitler Did you get a chance to test this at the observatory on your recent trip?

@AnthonyHorton is this something I could try next week as well?

**Pull request merged**